src/dashboards/components/dashboard_index/DashboardCardsPaginated.tsx

```diff
@@ -85,6 +86,14 @@ class DashboardCards extends PureComponent<OwnProps & StateProps> {
       </div>
     )
   }
+
+  public handlePinDashboard = () => {
+    getPinnedItems()
+      .then(res => {
+        this.setState({pinnedItems: res})
+      })
+      .catch(err => console.error(err))
+  }
```

Comment:
I would expected this function to be called in the `componentDidMount`

Reply:
Good catch! I was thinking of the same thing since I didn't want to duplicate code. I tried to call the `handlePinDashboard` in the `componentDidMount` and I kept getting this React warning saying:
```
Warning: Can't perform a React state update on an unmounted component. 
This is a no-op, but it indicates a memory leak in your application.
```

Here are the things I tried:

**Attempt 1**

```TypeScript
public componentDidMount() {
  if (isFlagEnabled('pinnedItems') && CLOUD) {
      if (this._isMounted) {
        this.handlePinDashboard()
      }
    }
}
```

**Attempt 2**

```TypeScript
public componentDidMount() {
  if (isFlagEnabled('pinnedItems') && CLOUD) {
    this.handlePinDashboard(this._isMounted)
  }
}

public handlePinDashboard = (isMounted: boolean) => {
  getPinnedItems()
    .then(res => {
      if (isMounted) {
        this.setState({pinnedItems: res})
      }
    })
    .catch(err => console.error(err))
}
```

Both implementations give the above warning. Seems like `this._isMounted` should be checked in the callback after `getPinnedItems()` is done, like [this](https://github.com/influxdata/ui/blob/f59f4b5e7b2502dc586b50494aae6d8b3fc818a5/src/dashboards/components/dashboard_index/DashboardCardsPaginated.tsx#L43). I'm not quite sure why this is the case.

I'm open to other suggestions!

Reply:
Ya, let's do that. The reason why is because the state can't be set on an unmounted component (as the error suggest). Since we only want to update the state when the component has been mounted, that's fine for our purposes.

Reply:
So the two attempts are both valid in our case even though the React warning keeps showing up?

Reply:
Sort of, you might want to try:


```
public componentDidMount() {
  if (isFlagEnabled('pinnedItems') && CLOUD) {
    this.handlePinDashboard()
  }
}

public handlePinDashboard = (isMounted: boolean) => {
 if (this._isMounted) {
  getPinnedItems()
    .then(res => {
      this.setState({pinnedItems: res})
    })
    .catch(err => console.error(err))
}
```